atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -265,106 +285,176 @@ private static boolean attemptToCreateKeyspaceOnHost(InetSocketAddress host, Cas
-        sanityCheckReplicationFactor(ks, config, dcs);
+    static void currentRfOnKeyspaceMatchesDesiredRf(
+            CassandraClient client, CassandraKeyspaceVerifierConfig cassandraKeyspaceVerifierConfig) throws TException {
+        KsDef ks = client.describe_keyspace(cassandraKeyspaceVerifierConfig.keyspace());
+        Set<String> dcs = sanityCheckDatacenters(
+                client,
+                cassandraKeyspaceVerifierConfig.servers(),
+                cassandraKeyspaceVerifierConfig.replicationFactor(),
+                cassandraKeyspaceVerifierConfig.ignoreNodeTopologyChecks());
+        sanityCheckReplicationFactor(
+                ks,
+                cassandraKeyspaceVerifierConfig.replicationFactor(),
+                dcs,
+                cassandraKeyspaceVerifierConfig.ignoreDatacenterConfigurationChecks());
     }
 
-    static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
+    static void sanityCheckReplicationFactor(
+            KsDef ks, int replicationFactor, Set<String> dcs, boolean ignoreDatacentreConfigurationChecks) {
         Set<String> scopedDownDcs = checkRfsSpecifiedAndScopeDownDcs(dcs, ks.getStrategy_options());
-        checkRfsMatchConfig(ks, config, scopedDownDcs, ks.getStrategy_options());
+        checkRfsMatchConfig(
+                ks, replicationFactor, scopedDownDcs, ks.getStrategy_options(), ignoreDatacentreConfigurationChecks);
     }
 
     private static Set<String> checkRfsSpecifiedAndScopeDownDcs(Set<String> dcs, Map<String, String> strategyOptions) {
         return Sets.intersection(dcs, strategyOptions.keySet());
     }
 
     private static void checkRfsMatchConfig(
-            KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs, Map<String, String> strategyOptions) {
+            KsDef ks,
+            int replicationFactor,
+            Set<String> dcs,
+            Map<String, String> strategyOptions,
+            boolean ignoreDatacenterConfigurationChecks) {
         for (String datacenter : dcs) {
-            if (Integer.parseInt(strategyOptions.get(datacenter)) != config.replicationFactor()) {
+            if (Integer.parseInt(strategyOptions.get(datacenter)) != replicationFactor) {
                 logErrorOrThrow(
                         "Your current Cassandra keyspace (" + ks.getName()
                                 + ") has a replication factor not matching your Atlas Cassandra configuration."
                                 + " Change them to match, but be mindful of what steps you'll need to"
                                 + " take to correctly repair or cleanup existing data in your cluster.",
-                        config.ignoreDatacenterConfigurationChecks());
+                        ignoreDatacenterConfigurationChecks);
             }
         }
     }
+
+    @Value.Immutable
+    interface CassandraKeyspaceVerifierConfig {
```

Comment:
Decided against using a staged builder here, given that this immutable is only build using the `of()` method outside of tests, and so there's currently little value added in making this staged.